test/system/workflow_executions/submissions_test.rb

```diff
@@ -2,37 +2,72 @@
 
 require 'application_system_test_case'
 
-class SubmissionsTest < ApplicationSystemTestCase
-  setup do
-    @user = users(:john_doe)
-    login_as @user
-    @sample1 = samples(:sample1)
-    @sample2 = samples(:sample2)
-    @project = projects(:project1)
-    @namespace = groups(:group_one)
-  end
+module WorkflowExecutions
+  class SubmissionsTest < ApplicationSystemTestCase
+    setup do
+      @user = users(:john_doe)
+      login_as @user
+      @sample1 = samples(:sample1)
+      @sample2 = samples(:sample2)
+      @project = projects(:project1)
+      @namespace = groups(:group_one)
+    end
 
-  test 'should display a pipeline selection modal' do
-    visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
+    test 'should display a pipeline selection modal for project samples' do
+      visit namespace_project_samples_url(namespace_id: @namespace.path, project_id: @project.path)
 
-    find("input[type='checkbox'][value='#{@sample1.id}']").click
-    find("input[type='checkbox'][value='#{@sample2.id}']").click
+      find("input[type='checkbox'][value='#{@sample1.id}']").click
+      find("input[type='checkbox'][value='#{@sample2.id}']").click
+
+      click_on I18n.t(:'projects.samples.index.workflows.button_sr')
+      within 'dialog[open]' do
+        assert_selector '.dialog--header', text: I18n.t(:'workflow_executions.submissions.pipeline_selection.title')
+        assert_button text: 'phac-nml/iridanextexample', count: 3
+        first('button', text: 'phac-nml/iridanextexample').click
+      end
 
-    click_on I18n.t(:'projects.samples.index.workflows.button_sr')
-    within 'dialog' do
-      assert_selector '.dialog--header', text: I18n.t(:'workflow_executions.submissions.pipeline_selection.title')
-      assert_button text: 'phac-nml/iridanextexample', count: 3
-      first('button', text: 'phac-nml/iridanextexample').click
+      within 'dialog[open].dialog--size-xl' do
+        within 'div.sample-sheet' do
+          within 'table' do
+            assert_selector 'tr[data-controller="nextflow--samplesheet"]', count: 2
+            assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample1.puid, count: 1
+            assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample2.puid, count: 1
+          end
+        end
+      end
     end
 
-    within 'dialog.dialog--size-xl' do
-      within 'div.sample-sheet' do
-        within 'table' do
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', count: 2
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample1.puid, count: 1
-          assert_selector 'tr[data-controller="nextflow--samplesheet"]', text: @sample2.puid, count: 1
+    test 'should display a pipeline selection modal for group samples' do
```

Comment:
For this test we should be logging in as an `analyst` to ensure that it works. I see the existing test also logged in as a user with an access level of `owner` which appeared to be missed

Reply:
I created a new group & project fixtures to test roles. ed16a029c2c9ee0cc099bf525a0edfc29f386376